### PR TITLE
Added response when search query returns no SO items

### PR DESCRIPTION
**Pull request by grobbie94**
Currently, stackit exits when search returns no stack overflow responses.
This just adds functionality to either launch browser into stack overflow, search again with new/edited terms or exit the program.

The os.system call to return to search isn't great, but any other way I could think of required complete restructure.

I'm pretty new to this, so feel free to be as critical as you want.

Thanks,
Rob

### CODE REVIEW

File: stackit/stackit_core.py [hunk 1]
```diff
@@ -77,7 +77,17 @@ def searchTerm(term, tags):
...
-
+    if len(questions) == 0:
+        response = raw_input("No questions found matching your input. Try broader search terms. \nPress s to search again, b to enter browser or x to exit: ")
+        if response == "x":
+            sys.exit()
+        elif response == "b":
+            webbrowser.open("https://stackoverflow.com")
+        elif response == "s":
+            searchTerms = raw_input("Enter search terms: ")
+            os.system('stackit %s' % searchTerms)
```
Comment (lukasschwab) [inline stackit/stackit_core.py]: I think the os.system call is a bit messy, and we could do this more cleanly.

@vickiniu could we do a recursive call to the `searchTerm` method here? Skips all the parsing stuff, and keeps it all in one process. Line 88 would become `searchTerm(searchTerms, "")`, unless we integrate tagging functionality.

File: stackit/stackit_core.py [hunk 2]
```diff
@@ -77,7 +77,17 @@ def searchTerm(term, tags):
...
+    
+
```
Comment (lukasschwab) [inline stackit/stackit_core.py]: Extra newline––it's nitpicky, but we'd love to see it gone.

File: stackit/stackit_core.py (hunk 1 above)
Comment (lukasschwab) [inline stackit/stackit_core.py]: Would the empty `tags` string look for empty tags, eliminating all results?

**Comment by lukasschwab** [comment]
Rob:

Thanks so much for the pull request! Several of us (me included) are super new to this too. I made some line comments in your code, and we'd love to see those changes in a new commit!

We agree that there should be some way to handle result-less queries, but I'm not sure the best way to handle that is to drop the user into an interface (the one you've created) that tries to recreate the functionality at the command line––that seems to be two ways to do functionally the same thing, start from scratch.

Additionally, because it bypasses the command line parsing process, dropping the user into such an interface could lose some functionality (such as tags or stderr piping) when you remove the messy `os.system()` call.

This is something we'll talk about as a team, and get back to you ASAP!

Just some brainstorming:
- What if we just print some sort of 'search failed' warning and then return the user to the command line?
- We should give the user the search terms––that already happens before listing the search results, and before this condition could be triggered, so this shouldn't be an issue.
- I want to print all error warnings in red––making this a Trello card.

Thanks again!

**Comment by lukasschwab** [comment]
@grobbie94 I just merged in another pull request, which means you're also going to have to rebase onto master. Just a heads up!

**Comment by grobbie94** [comment]
Hey Lukas,

Removed the option to 're-search' with new terms, that does indeed take away some of the main functionality of piping in errors. 

Now, a warning comes up with the search terms used. If the user wants to enter the browser, it directs to stackoverflow with the search completed so it can be easily altered.

**Comment by lukasschwab** [comment]
@grobbie94 something about your `git rebase` seems a bit odd–-see the 'Files Changed' tab on this pull request.

The `master` code has changed quite a bit since this pull request was opened, including major refactoring, which I think would make the continued modification of this pull request tricky. My suggestion is that you watch [PR 9: Refactoring](https://github.com/lukasschwab/stackit/pull/9), re-branch when that pull request is merged, and then re-implement your functionality.

If you aren't liking this, I'd be happy to implement it myself and give you credit for this pull request conversation in README.md.

Just let me know!

**Comment by grobbie94** [comment]
@lukasschwab I haven't been watching very closely. I'll re fork it and add it in.

Thanks,

Rob

**Pull request closed without merging**